gcp_variant_transforms/testing/temp_dir.py

```diff
@@ -40,20 +44,37 @@ def get_path(self):
     """Returns the path to the temporary directory."""
     return self._tempdir
 
-  def create_temp_file(self, suffix='', lines=None):
+  def create_temp_file(
+      self, suffix='', lines=None,
+      compression_type=filesystem.CompressionTypes.UNCOMPRESSED):
     """Creates a temporary file in the temporary directory.
+
     Args:
       suffix (str): The filename suffix of the temporary file (e.g. '.txt')
       lines (List[str]): A list of lines that will be written to the temporary
         file.
+      compression_type (str): Specifies compression type of the file. Value
+        should be one of ``CompressionTypes``.
     Returns:
       The name of the temporary file created.
+    Raises:
+      ValueError: If ``compression_type`` is unsupported.
     """
     f = tempfile.NamedTemporaryFile(delete=False,
                                     dir=self._tempdir,
                                     suffix=suffix)
-    if lines:
-      for line in lines:
-        f.write(line)
+    if not lines:
+      return f.name
+    if compression_type in (filesystem.CompressionTypes.UNCOMPRESSED,
+                            filesystem.CompressionTypes.AUTO):
+      f.write(''.join(lines))
```

Comment:
I wonder if we should actually change this to always append new lines in a standard way, before running anything we could just do `lines = '\n'.join([line.strip() for line in lines]).

Reply:
Actually, part of the tests were to test '\n', and '\r\n', so I think it's ok to leave as is.